vortexasdk/result_conversions.py

```diff
@@ -14,6 +14,15 @@ def create_list(list_of_dicts, output_class: FromDictMixin) -> List:
     return [output_class.from_dict(d) for d in list_of_dicts]
 
 
+def format_datatypes(df: pd.DataFrame) -> pd.DataFrame:
+    """Format the relevant columns with sensible datatypes"""
+    timestamp_cols = [c for c in df.columns if "timestamp" in c]
+
+    df[timestamp_cols] = df[timestamp_cols].apply(pd.to_datetime)
```

Comment:
Pandas does not allow you to set multi-column slices like this (see example below). You need to do:

for timestamp_col in timestamp_cols:
    df[timestamp_col] = pd.to_datetime(df[timestamp_col])

example:
a = pd.DataFrame({'a': [0, 1], 'b': [2, 3], 'c': [4, 5]})
a[['a', 'b']] = a[['a', 'b']] + 1
print(a)
   a  b  c
0  1  3  4
1  2  4  5

Reply:
Using the apply method does allow this - it's a different implementation to `a[['a', 'b']] = a[['a', 'b']] + 1`.

I've added a test though to confirm this works.

Reply:
Note on this - Although it incurs the overhead of a loop, a more efficient implementation is probably:

```python
for col in timetstamp_cols:
    pd.to_datetime(df[c])
```

As you now no longer have the apply overhead and applies on dataframes generally aren't all that efficient (checked on a local sample df, this is about 5x faster than the df apply)